### PR TITLE
payments controller and the interface

### PR DESCRIPTION
**Pull request by michaelt16**

### CODE REVIEW

File: src/controllers/PaymentsController.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import {Payments, IPayments} from "../database/models/Payments";
+
+export default class PaymentsContrtoller {
```
Comment (dingdongg) [inline src/controllers/PaymentsController.ts]: typo, should be "Controller"

File: src/database/models/Payments.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+
+
+enum Status {
+    Accepted,
+    Pending,
+    Rejected
+} 
+
+export interface IPayments {
```
Comment (dingdongg) [inline src/database/models/Payments.ts]: I think we should make the name non-plural. The name implies that this object holds a list of payments when the object only represents a single payment.

File: src/database/models/Payments.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    memberId: Schema.Types.ObjectId,
+    payment:string,
```
Comment (dingdongg) [inline src/database/models/Payments.ts]: what does `payment` represent?

File: src/database/models/Payments.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { Schema, model } from "mongoose";
+
+
+
+enum Status {
```
Comment (dingdongg) [inline src/database/models/Payments.ts]: we can create a separate `enums` folder and create a file for the `Status` enum, that way we can reduce cluttering in the files

Comment (dingdongg) [reply src/database/models/Payments.ts]: when you do that, you also probably want to create functions that can convert from enum to string (and vice versa) so we can easily store on the DB

File: src/services/Schedueler.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (dingdongg) [inline src/services/Schedueler.ts:2]: typo in file name

**Review by dingdongg — CHANGES_REQUESTED** [review]
made some comments

File: src/database/models/Payments.ts (hunk 3 above)
Comment (michaelt16) [reply src/database/models/Payments.ts]: it was a stub, i edited it now.

File: src/database/models/Payments.ts (hunk 4 above)
Comment (michaelt16) [reply src/database/models/Payments.ts]: yup toString() works, and i created a seperate enum folder as well

**Comment by michaelt16** [comment]
New and updated pull request.

- Added cron functionality called cronPayment() under UserController.ts

- Added two functions that builds payment history under PaymentController.ts, the createPaymentHistory and findUserHistory(userId)

- Updated schema so that it has the memberId paying for the payment.

I will make tests for these functions once its merged using mocha,chai

File: src/database/models/Payments.ts [hunk 6]
```diff
@@ -1,18 +1,16 @@
...
     paymentAmount: Number,
     paymentDate:Date,
     paymentStatus: String;
 
 }
 
-
 const paymentsSchema = new Schema <IPayment>({
-    memberId: { type: Schema.Types.ObjectId, required: false},
+    memberId: { type: Schema.Types.ObjectId, required: true },
```
Comment (dingdongg) [inline src/database/models/Payments.ts]: `
memberId: { type: Schema.Types.ObjectId, ref: "User" },
`

**Review by dingdongg — APPROVED** [review]
lgtm `2979abc73a110ac775ede2cae01d644472a3a229`

**Pull request merged**